sky/clouds/service_catalog/aws_catalog.py

```diff
@@ -31,58 +35,79 @@
 # skypilot-catalog/.github/workflows/update-aws-catalog.yml
 _PULL_FREQUENCY_HOURS = 7
 
-_df = common.read_catalog('aws/vms.csv',
-                          pull_frequency_hours=_PULL_FREQUENCY_HOURS)
+_default_df = common.read_catalog('aws/vms.csv',
+                                  pull_frequency_hours=_PULL_FREQUENCY_HOURS)
+_user_df = None
+_apply_az_mapping_lock = threading.Lock()
+
 _image_df = common.read_catalog('aws/images.csv',
                                 pull_frequency_hours=_PULL_FREQUENCY_HOURS)
 
 
 def _apply_az_mapping(df: pd.DataFrame) -> pd.DataFrame:
     """Maps zone IDs (use1-az1) to zone names (us-east-1x).
 
+    The caller should guarantee that the aws credentials are valid.
+
     Such mappings are account-specific and determined by AWS. We fetch the
     mappings from AWS, which requires AWS credentials. If the user does not
-    have AWS credentials configured, we use zone id as the zone name, which
-    is only used for show-gpus. It is ok to use the default mapping because
-    the user will not be able to provision instances with those wrong
-    availablity zones due to the lack of credentials.
+    have AWS credentials configured, we use zone name mapping of the
+    SkyPilot's dev account (or zone id directly due to no zone name in an
+    older version). It is ok to use the default mapping because the user
+    will not be able to provision instances with those wrong availablity
+    zones due to the lack of credentials.
 
-    The mappings will also serve to remove from '_df' the regions that are
+    The mappings will also serve to remove from 'df' the regions that are
     not supported by the user account.
 
     Returns:
         A dataframe with column 'AvailabilityZone' that's correctly replaced
         with the zone name (e.g. us-east-1a).
     """
-    aws_enabled, _ = aws.AWS().check_credentials()
-    if aws_enabled:
-        user_identity = aws.AWS().get_current_user_identity()
+    # The caller should guarantee that the aws credentials are valid.
+    try:
+        user_identity = aws.AWS.get_current_user_identity()
         assert user_identity is not None, 'user_identity is None'
         aws_user_hash = hashlib.md5(user_identity.encode()).hexdigest()[:8]
-    else:
-        aws_user_hash = 'default'
+    except exceptions.CloudUserIdentityError:
+        glob_name = common.get_catalog_path('aws/az_mappings-*.csv')
+        # Find the most recent file that matches the glob.
+        glob_files = glob.glob(glob_name)
+        if glob_files:
+            glob_files.sort(key=os.path.getmtime)
+            aws_user_hash = os.path.basename(glob_files[-1]).split('-')[1]
+            aws_user_hash = aws_user_hash.split('.')[0]
+        else:
+            aws_user_hash = 'default'
```

Comment:
Curious why we find the latest file, vs. just use `default`? Document the reason?